pkg/image/validation/validation.go

```diff
@@ -15,6 +15,7 @@ func ValidateDefinition(ctx *image.Context) map[string][]FailedValidation {
 	failures := map[string][]FailedValidation{}
 
 	validations := map[string]validateComponent{
+		versionComponent:  validateVersion,
```

Comment:
My gut reaction was that this is some sort of pre-step before the other validations, since it doesn't make sense to run them if there's a version incompatibility. But seeing this, I think it makes sense. In the future this may lead to some confusing validation failure messages since a failure doesn't stop subsequent checks, but we'll burn that bridge when we get to it.